src/subgraph/concatenate.c

```diff
@@ -149,7 +116,7 @@ static enum xnn_status reshape_concatenate_n_operator(
 {
   enum xnn_status status;
 
-  assert(opdata->num_inputs == num_inputs);
+  num_inputs = opdata->num_inputs;
```

Comment:
There needs to be an error check here, that `num_inputs <= XNN_MAX_OPERATOR_OJBECTS`